src/client/index.ts

```diff
@@ -23,24 +24,27 @@ export function ViteSSG(
     rootContainer = '#app',
     hydration = false,
   } = options
-  const isClient = typeof window !== 'undefined'
 
-  async function createApp(client = false, routePath?: string) {
-    const app = client && !hydration
-      ? createClientApp(App)
-      : createSSRApp(App)
+  async function createApp(_client = false, routePath?: string) {
+    const app = import.meta.env.SSR || hydration
```

Comment:
maybe we should add a warning here if calling `createApp(false)` with `!import.meta.env.SSR` (some for spa)

Reply:
This or adding more context in release notes about misusage, as it should not make sense to call `createApp` with `client: false` in a non-client context